wolfcrypt/src/port/Renesas/renesas_rx64_hw_sha.c

```diff
@@ -0,0 +1,424 @@
+/* renesas_rx64_hw_sha.c
+ *
+ * Copyright (C) 2020-2022 Johnson Controls International Plc
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include <string.h>
+#include <stdio.h>
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+#include <wolfssl/wolfcrypt/settings.h>
+
+#if !defined(NO_SHA) || !defined(NO_SHA256)
+
+#include <wolfssl/wolfcrypt/logging.h>
+
+#if defined(WOLFSSL_RENESAS_RX64_HW)
```

Comment:
RX64 H/W acceleration seems to only support ShaXXX at the moment, but other features may be added in the future. Just in case, instead of defining WOLFSSL_RENESAS_RX64_HW,
WOLFSSL_RENESAS_RX64_SHA or WOLFSSL_RENESAS_RX64_HASH may be better.
Also, it seems that there is currently a limit to this accelerator, but I think it will be necessary to switch processing once the limit is lifted. At that time, it may be better to define the version number of the H/W driver (FIT module). For example, if the version of this module is 1.23:
#define WOLFSSL_RENESAS_RX64_FITVER 123

Reply:
So it would be preferred to change 
```c
#if defined(WOLFSSL_RENESAS_RX64_HW)
```
to 
```c
#if defined(WOLFSSL_RENESAS_RX64_HASH)
```

How would the version define work? Would that be used in place of either of the above defines?
For reference we've been using version 1.01

Reply:
Renesas TSIP sometimes added arguments with the same function name. Therefore, it was necessary to call the function differently depending on the version of TSIP. Also, if you're writing code to work around a certain RX64M H/W feature limitation, you'll likely need to switch to another code in the future when the limitation is lifted.
The use case would be like as follows:

```
#define RX64_DRIVER_VER  102    // means ver 1.02

/* foo has limitation in h/w driver ver 1.00 but resolved in ver 1.01 */
int feature_foo()
{
     #if (RX64_DRIVER_VER <= 101)
            Do work around...
     #else
            Go straight forward code...
     #endif
}
```

Reply:
I've updated it to use WOLFSSL_RENESAS_RX64_HASH.

Regarding the version limitations. It looks like version 1.01 is the most current and it's dated early 2015 so I don't think a future version is coming that is going to resolve any of the limitations of this API. It seems like TSIP is what Renesas is using in all of it's new processors and that is seeing all of the active development instead.

Reply:
Thanks for the update.
I have heard that Renesas updates H/W accelerators depending on the size of user requests (business scale). Similar update opportunities are possible for RX64. However, the decision to introduce version numbers could be deferred until that future update.